domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -444,23 +473,60 @@ class TopicListController @Inject constructor(
     topicProgressList: List<TopicProgress>
   ): List<PromotedStory> {
     val recommendedStories = mutableListOf<PromotedStory>()
-    topicProgressList.forEach {
-      val dependentListOfTopics =
-        getDependentLisOfTopicForOngoingTopic(it.topicId)
-      for (j in dependentListOfTopics.indices) {
-        val found = topicProgressList.any { it.topicId == dependentListOfTopics[j] }
-        if (!found) {
-          val recommendedStoriesIdFromAssets =
-            createRecommendedStoryFromAssets(dependentListOfTopics[j])
-          if (recommendedStoriesIdFromAssets != null) {
-            recommendedStories.add(recommendedStoriesIdFromAssets)
+    val topicIdJsonArray = jsonAssetRetriever
+      .loadJsonFromAsset("topics.json")!!
+      .getJSONArray("topic_id_list")
+
+    // Check if any prerequisite topic user needs to learn.
+    for (i in 0 until topicIdJsonArray.length()) {
+      val topicInProgressFound = topicProgressList.any { it.topicId == topicIdJsonArray[i] }
+      if (topicInProgressFound) {
+        val dependentListOfTopics =
+          retrieveTopicDependencies(topicIdJsonArray[i].toString())
+        for (j in dependentListOfTopics.indices) {
+          val dependentTopicIdsFound =
+            topicProgressList.any { it.topicId == dependentListOfTopics[j] }
+          val recommendedTopicAlreadyExist =
+            recommendedStories.any { it.topicId == dependentListOfTopics[j] }
+          if (!dependentTopicIdsFound && !recommendedTopicAlreadyExist) {
+            fetchListOfRecommendedStories(dependentListOfTopics[j])?.let {
+              recommendedStories.add(it)
+            }
+          }
+        }
+      }
+    }
+
+    // If no any dependent topics to recommend. Recommend next topic,only if any ongoing topic is completed.
```

Comment:
I think with the adjustments I suggested in my comment above this block of code may not be needed.

Reply:
Removed.